### DISCUSSION TITLE
登录页支持 Turnstile 人机验证

### DISCUSSION THREAD

**Discussion by blhcm** [Qexo/Qexo#516] [Ideas]
如题，建议在面板 设置-安全配置 一项增加对 Cloudflare Turnstile 的支持，Cloudflare Turnstile 是一款替代 CAPTCHA 的免费工具，在中国大陆的体验可能比 reCAPTCHA 更好一些。

https://www.cloudflare.com/products/turnstile/

**Comment by yzl3014** [comment]
reCAPTCHA v2/v3 现已[不可用](https://developers.google.com/recaptcha/intro?hl=zh-cn)，新版早在2024年就已开始[涨价](https://www.ithome.com/0/748/120.htm)。刚刚我想登录Qexo，结果一直提示人机验证不通过，我只能从数据库里删了API Key，然后让Vercel重新部署才解决。

用 Cloudflare Turnstile 作为代替: https://developers.cloudflare.com/turnstile/get-started/

**Reply by am-abudu** [reply]
reCAPTCHA 免费额度仍然可用, 人机验证不通过可能是因为网络问题。Cloudflare Turnstile 将会提上日程
